ppc64le: Massive warnings with '-Wnarrowing' flag enabled

<!--
If you have a question rather than reporting a bug please go to http://answers.opencv.org where you get much faster responses.
If you need further assistance please read [How To Contribute](https://github.com/opencv/opencv/wiki/How_to_contribute).

Please:

* Read the documentation to test with the latest developer build.
* Check if other person has already created the same issue to avoid duplicates. You can comment on it if there already is an issue.
* Try to be as detailed as possible in your report.
* Report only one problem per created issue.


This is a template helping you to create an issue which can be processed as quickly as possible. This is the bug reporting section for the OpenCV library.
-->

##### System information (version)
<!-- Example
- OpenCV => 3.1
- Operating System / Platform => Windows 64 Bit
- Compiler => Visual Studio 2015
-->

- OpenCV => 3.4 and master
- Operating System / Platform =>  ppc64le GNU/Linux
- Compiler => gcc

##### Detailed description
```
/worker/buildbot/Power9_Linux_gcc-6__opencv/opencv/modules/core/include/opencv2/core/hal/intrin_vsx.hpp: In function 'cv::hal_baseline::v_uint8x16 cv::hal_baseline::operator<<(const cv::hal_baseline::v_uint8x16&, int)':
/worker/buildbot/Power9_Linux_gcc-6__opencv/opencv/modules/core/include/opencv2/core/vsx_utils.hpp:18:77: warning: narrowing conversion of 'imm' from 'int' to 'unsigned char' inside { } [-Wnarrowing]
 #define __VSX_S16__(c, v) (c){v, v, v, v, v, v, v, v, v, v, v, v, v, v, v, v}
                                                                             ^
/worker/buildbot/Power9_Linux_gcc-6__opencv/opencv/modules/core/include/opencv2/core/vsx_utils.hpp:25:31: note: in expansion of macro '__VSX_S16__'
 #define vec_uchar16_sp(c)    (__VSX_S16__(vec_uchar16, c))
                               ^~~~~~~~~~~
/worker/buildbot/Power9_Linux_gcc-6__opencv/opencv/modules/core/include/opencv2/core/hal/intrin_vsx.hpp:533:31: note: in expansion of macro 'vec_uchar16_sp'
 { return _Tpvec(vec_sl(a.val, splfunc(imm))); }              \
                               ^~~~~~~
/worker/buildbot/Power9_Linux_gcc-6__opencv/opencv/modules/core/include/opencv2/core/hal/intrin_vsx.hpp:541:1: note: in expansion of macro 'OPENCV_HAL_IMPL_VSX_SHIFT_OP'
 OPENCV_HAL_IMPL_VSX_SHIFT_OP(v_uint8x16, vec_sr, vec_uchar16_sp)
 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/worker/buildbot/Power9_Linux_gcc-6__opencv/opencv/modules/core/include/opencv2/core/vsx_utils.hpp:18:77: warning: narrowing conversion of 'imm' from 'int' to 'unsigned char' inside { } [-Wnarrowing]
 #define __VSX_S16__(c, v) (c){v, v, v, v, v, v, v, v, v, v, v, v, v, v, v, v}
                                                                             ^
/worker/buildbot/Power9_Linux_gcc-6__opencv/opencv/modules/core/include/opencv2/core/vsx_utils.hpp:25:31: note: in expansion of macro '__VSX_S16__'
 #define vec_uchar16_sp(c)    (__VSX_S16__(vec_uchar16, c))
                               ^~~~~~~~~~~
/worker/buildbot/Power9_Linux_gcc-6__opencv/opencv/modules/core/include/opencv2/core/hal/intrin_vsx.hpp:533:31: note: in expansion of macro 'vec_uchar16_sp'
 { return _Tpvec(vec_sl(a.val, splfunc(imm))); }              \
                               ^~~~~~~
....
```


CI: https://ocv-power.imavr.com/#/builders/5/builds/23

##### Steps to reproduce

build OpenCV on ppc64le

--- a/modules/core/include/opencv2/core/vsx_utils.hpp
+++ b/modules/core/include/opencv2/core/vsx_utils.hpp
@@ -22,37 +22,37 @@
 
 typedef __vector unsigned char vec_uchar16;
 #define vec_uchar16_set(...) (vec_uchar16){__VA_ARGS__}
-#define vec_uchar16_sp(c)    (__VSX_S16__(vec_uchar16, c))
+#define vec_uchar16_sp(c)    (__VSX_S16__(vec_uchar16, (unsigned char)c))
 #define vec_uchar16_c(v)     ((vec_uchar16)(v))
 #define vec_uchar16_z        vec_uchar16_sp(0)
 
 typedef __vector signed char vec_char16;
 #define vec_char16_set(...) (vec_char16){__VA_ARGS__}
-#define vec_char16_sp(c)    (__VSX_S16__(vec_char16, c))
+#define vec_char16_sp(c)    (__VSX_S16__(vec_char16, (signed char)c))
 #define vec_char16_c(v)     ((vec_char16)(v))
 #define vec_char16_z        vec_char16_sp(0)
 
 typedef __vector unsigned short vec_ushort8;
 #define vec_ushort8_set(...) (vec_ushort8){__VA_ARGS__}
-#define vec_ushort8_sp(c)    (__VSX_S8__(vec_ushort8, c))
+#define vec_ushort8_sp(c)    (__VSX_S8__(vec_ushort8, (unsigned short)c))
 #define vec_ushort8_c(v)     ((vec_ushort8)(v))
 #define vec_ushort8_z        vec_ushort8_sp(0)
 
 typedef __vector signed short vec_short8;
 #define vec_short8_set(...) (vec_short8){__VA_ARGS__}
-#define vec_short8_sp(c)    (__VSX_S8__(vec_short8, c))
+#define vec_short8_sp(c)    (__VSX_S8__(vec_short8, (signed short)c))
 #define vec_short8_c(v)     ((vec_short8)(v))
 #define vec_short8_z        vec_short8_sp(0)
 
 typedef __vector unsigned int vec_uint4;
 #define vec_uint4_set(...) (vec_uint4){__VA_ARGS__}
-#define vec_uint4_sp(c)    (__VSX_S4__(vec_uint4, c))
+#define vec_uint4_sp(c)    (__VSX_S4__(vec_uint4, (unsigned int)c))
 #define vec_uint4_c(v)     ((vec_uint4)(v))
 #define vec_uint4_z        vec_uint4_sp(0)
 
 typedef __vector signed int vec_int4;
 #define vec_int4_set(...)  (vec_int4){__VA_ARGS__}
-#define vec_int4_sp(c)     (__VSX_S4__(vec_int4, c))
+#define vec_int4_sp(c)     (__VSX_S4__(vec_int4, (signed int)c))
 #define vec_int4_c(v)      ((vec_int4)(v))
 #define vec_int4_z         vec_int4_sp(0)
 
@@ -64,13 +64,13 @@ typedef __vector float vec_float4;
 
 typedef __vector unsigned long long vec_udword2;
 #define vec_udword2_set(...) (vec_udword2){__VA_ARGS__}
-#define vec_udword2_sp(c)    (__VSX_S2__(vec_udword2, c))
+#define vec_udword2_sp(c)    (__VSX_S2__(vec_udword2, (unsigned long long)c))
 #define vec_udword2_c(v)     ((vec_udword2)(v))
 #define vec_udword2_z        vec_udword2_sp(0)
 
 typedef __vector signed long long vec_dword2;
 #define vec_dword2_set(...) (vec_dword2){__VA_ARGS__}
-#define vec_dword2_sp(c)    (__VSX_S2__(vec_dword2, c))
+#define vec_dword2_sp(c)    (__VSX_S2__(vec_dword2, (signed long long)c))
 #define vec_dword2_c(v)     ((vec_dword2)(v))
 #define vec_dword2_z        vec_dword2_sp(0)
 

